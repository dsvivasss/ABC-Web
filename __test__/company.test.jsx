import React from 'react';
import { render } from '@testing-library/react';
import Company from '../src/app/company/page';
import '@testing-library/jest-dom';

// Mock useRouter:
jest.mock("next/navigation", () => ({
    useRouter() {
      return {
        prefetch: () => null
      };
    }
  }));

describe('Company page', () => {
    it('Company page renders correctly', () => {
        const { container } = render(<Company />)
        expect(container).toMatchSnapshot()
    });
})