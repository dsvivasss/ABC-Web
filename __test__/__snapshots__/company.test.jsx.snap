// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company page Company page renders correctly 1`] = `
<div>
  <main
    class="main"
  >
    <div
      class="center"
    >
      <h1
        class="animate-fade-up from-black bg-clip-text text-center font-display text-4xl font-bold tracking-[-0.02em] [text-wrap:balance] md:text-7xl md:leading-[5rem]"
        style="animation-delay: 0.15s; animation-fill-mode: forwards;"
      >
        Registro
      </h1>
    </div>
    <div
      class="mx-auto mt-6 flex animate-fade-up items-center justify-center space-x-5 opacity-0 pb-6"
      style="animation-delay: 0.3s; animation-fill-mode: forwards;"
    >
      <a
        class="flex max-w-fit items-center justify-center space-x-2 rounded-full border border-gray-300 bg-white px-5 py-2 text-sm text-gray-600 shadow-md transition-colors hover:border-gray-800"
        href="/"
        rel="noopener noreferrer"
      >
        <p>
          Accede como aspirante
        </p>
      </a>
      <a
        class="group flex max-w-fit items-center justify-center space-x-2 rounded-full border border-black bg-black px-5 py-2 text-sm text-white transition-colors hover:bg-white hover:text-black"
        href="/company"
        rel="noopener noreferrer"
      >
        <p>
          <span
            class="hidden sm:inline-block"
          >
            Accede como empresa
          </span>
        </p>
      </a>
    </div>
    <div
      class="card"
    >
      <div>
        <h2>
          Conectamos profesionales de TI
        </h2>
      </div>
      <div
        class="mt-10 sm:mx-auto sm:w-full sm:max-w-sm"
      >
        <form
          action="#"
          class="space-y-6"
          method="POST"
        >
          <div>
            <label
              class="block text-sm font-medium leading-6 text-gray-900"
              for="company_name"
            >
              Nombre de la empresa
            </label>
            <div
              class="mt-2"
            >
              <input
                autocomplete="company_name"
                class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                id="company_name"
                name="company_name"
                placeholder="  Nombre de la empresa"
                required=""
                type="company_name"
              />
            </div>
          </div>
          <div>
            <label
              class="block text-sm font-medium leading-6 text-gray-900"
              for="email"
            >
              Correo del representante
            </label>
            <div
              class="mt-2"
            >
              <input
                autocomplete="email"
                class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                id="email"
                name="email"
                placeholder="name@domain.com"
                required=""
                type="email"
              />
            </div>
          </div>
          <div>
            <div
              class="flex items-center justify-between"
            >
              <label
                class="block text-sm font-medium leading-6 text-gray-900"
                for="password"
              >
                Contraseña
              </label>
            </div>
            <div
              class="mt-2"
            >
              <input
                autocomplete="current-password"
                class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                id="password"
                name="password"
                placeholder="********"
                required=""
                type="password"
              />
            </div>
          </div>
          <label
            class="block text-sm font-medium leading-6 text-gray-900"
            data-headlessui-state=""
            id="headlessui-listbox-label-:r0:"
          >
            Tamaño de la empresa
          </label>
          <div
            class="relative mt-2"
          >
            <button
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="headlessui-listbox-label-:r0: headlessui-listbox-button-:r1:"
              class="relative w-full cursor-default rounded-md bg-white py-1.5 pr-10 text-left text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 focus:outline-none focus:ring-2 focus:ring-indigo-500 sm:text-sm sm:leading-6"
              data-headlessui-state=""
              id="headlessui-listbox-button-:r1:"
              type="button"
            >
              <span
                class="flex items-center"
              >
                <span
                  class="ml-3 block truncate"
                >
                  &lt; 100 empleados
                </span>
              </span>
              <span
                class="pointer-events-none absolute inset-y-0 right-0 ml-2 flex items-center pr-2 pt-2"
              >
                <svg
                  aria-hidden="true"
                  class="h-5 w-5 text-gray-900"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M10 3a.75.75 0 01.55.24l3.25 3.5a.75.75 0 11-1.1 1.02L10 4.852 7.3 7.76a.75.75 0 01-1.1-1.02l3.25-3.5A.75.75 0 0110 3zm-3.76 9.2a.75.75 0 011.06.04l2.7 2.908 2.7-2.908a.75.75 0 111.1 1.02l-3.25 3.5a.75.75 0 01-1.1 0l-3.25-3.5a.75.75 0 01.04-1.06z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
          <label
            class="block text-sm font-medium leading-6 text-gray-900"
            data-headlessui-state=""
            id="headlessui-listbox-label-:r2:"
          >
            Ubicación
          </label>
          <div
            class="relative mt-2"
          >
            <button
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="headlessui-listbox-label-:r2: headlessui-listbox-button-:r3:"
              class="relative w-full cursor-default rounded-md bg-white py-1.5 pr-10 text-left text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 focus:outline-none focus:ring-2 focus:ring-indigo-500 sm:text-sm sm:leading-6"
              data-headlessui-state=""
              id="headlessui-listbox-button-:r3:"
              type="button"
            >
              <span
                class="flex items-center"
              >
                <span
                  class="ml-3 block truncate"
                >
                  Colombia
                </span>
              </span>
              <span
                class="pointer-events-none absolute inset-y-0 right-0 ml-2 flex items-center pr-2 pt-2"
              >
                <svg
                  aria-hidden="true"
                  class="h-5 w-5 text-gray-900"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M10 3a.75.75 0 01.55.24l3.25 3.5a.75.75 0 11-1.1 1.02L10 4.852 7.3 7.76a.75.75 0 01-1.1-1.02l3.25-3.5A.75.75 0 0110 3zm-3.76 9.2a.75.75 0 011.06.04l2.7 2.908 2.7-2.908a.75.75 0 111.1 1.02l-3.25 3.5a.75.75 0 01-1.1 0l-3.25-3.5a.75.75 0 01.04-1.06z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
          <label
            class="block text-sm font-medium leading-6 text-gray-900"
            data-headlessui-state=""
            id="headlessui-listbox-label-:r4:"
          >
            Sector
          </label>
          <div
            class="relative mt-2"
          >
            <button
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="headlessui-listbox-label-:r4: headlessui-listbox-button-:r5:"
              class="relative w-full cursor-default rounded-md bg-white py-1.5 pr-10 text-left text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 focus:outline-none focus:ring-2 focus:ring-indigo-500 sm:text-sm sm:leading-6"
              data-headlessui-state=""
              id="headlessui-listbox-button-:r5:"
              type="button"
            >
              <span
                class="flex items-center"
              >
                <span
                  class="ml-3 block truncate"
                >
                  Tecnología
                </span>
              </span>
              <span
                class="pointer-events-none absolute inset-y-0 right-0 ml-2 flex items-center pr-2 pt-2"
              >
                <svg
                  aria-hidden="true"
                  class="h-5 w-5 text-gray-900"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M10 3a.75.75 0 01.55.24l3.25 3.5a.75.75 0 11-1.1 1.02L10 4.852 7.3 7.76a.75.75 0 01-1.1-1.02l3.25-3.5A.75.75 0 0110 3zm-3.76 9.2a.75.75 0 011.06.04l2.7 2.908 2.7-2.908a.75.75 0 111.1 1.02l-3.25 3.5a.75.75 0 01-1.1 0l-3.25-3.5a.75.75 0 01.04-1.06z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div>
            <label
              class="block text-sm font-medium leading-6 text-gray-900"
              for="website"
            >
              Sitio web
            </label>
            <div
              class="mt-2"
            >
              <input
                autocomplete="website"
                class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                id="website"
                name="website"
                placeholder="  Ingresa el sitio web de la empresa"
                required=""
                type="website"
              />
            </div>
          </div>
          <div
            class="pb-1"
          >
            <button
              class="flex w-full justify-center rounded-md bg-indigo-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
              href="https://nextjs.org/learn?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
              type="submit"
            >
              Registrarse
            </button>
          </div>
        </form>
        <p
          class="mt-10 text-center text-sm text-gray-500 pt-4"
        >
          ¿Ya tienes una cuenta?
           
          <a
            class="font-semibold leading-6 text-indigo-600 hover:text-indigo-500"
            href="https://nextjs.org/learn?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
          >
            Iniciar sesión
          </a>
        </p>
      </div>
    </div>
    <div
      class="Toastify"
    />
  </main>
</div>
`;
